health-events-analyzer/pkg/publisher/publisher.go

```diff
@@ -93,13 +95,19 @@ func NewPublisher(platformConnectorClient pb.PlatformConnectorClient) *Publisher
 }
 
 func (p *PublisherConfig) Publish(ctx context.Context, event *pb.HealthEvent,
-	recommendedAction pb.RecommenedAction) error {
-	// Create the health events request
-	event.IsFatal = true
-	event.RecommendedAction = recommendedAction
+	recommendedAction pb.RecommenedAction, ruleName string) error {
+	newEvent := proto.Clone(event).(*pb.HealthEvent)
+
+	// Override fields with new values
+	newEvent.Agent = "health-events-analyzer"
+	newEvent.CheckName = ruleName                  // change the check name to HealthEventsAnalyzer
```

Comment:
This comment is not quite correct. We're changing to check name to the rule name, not HealthEventAnalyser.